recipe/meta.yaml

```diff
@@ -22,13 +23,61 @@ requirements:
     # compiler on osx depends on zlib itself. Use the bootstrap compiler instead on osx
     - {{ compiler('c') }}
     - make                  # [unix]
+  run:
+
+outputs:
+  - name: libzlib
+    build:
+      run_exports:
+        # mostly OK, but some scary symbol removal.  Let's try for trusting them.
+        #  https://abi-laboratory.pro/tracker/timeline/zlib/
+        - {{ pin_subpackage('libzlib', max_pin='x.x') }}
+    requirements:
+      build:
+        - {{ compiler('c') }}
+      host:
+      run:
+      run_constrained:
+        - zlib {{ version }} *_{{ build_num }}
```

Comment:
Can we make this just
```yaml
- {{ pin_subpackage('zlib', exact=True) }}
```
? Since the `zlib` output has `{{ pin_subpackage('libzlib', exact=True) }}` then all builds that satisfy the constraint should be those from `{{ pin_subpackage('zlib', exact=True) }}` either way (and we shouldn't care about build from other channels when comparing build numbers).

Reply:
Nope. It's a chicken and egg problem. conda-build doesn't know about the PKG_HASH of `zlib` until `libzlib` has been rendered, but rendering `libzlib` needs the PKG_HASH of `zlib` if we use `exact=True`.